backend/drm/util.c

```diff
@@ -9,7 +9,7 @@
 #include "backend/drm/util.h"
 
 int32_t calculate_refresh_rate(const drmModeModeInfo *mode) {
-	int32_t refresh = (mode->clock * 1000000LL / mode->htotal +
+	int32_t refresh = (((int64_t) mode->clock) * 1000000LL / mode->htotal +
```

Comment:
```suggestion
	uint64_t refresh = ((mode->clock) * 1000000LL / mode->htotal +
```
This will also require adjustments to `struct wlr_output_mode`.

Reply:
This is only required if the above calculation exceeds `INT32_MAX`.

Reply:
Asked on dri-devel: https://lists.freedesktop.org/archives/dri-devel/2020-October/283935.html

Reply:
These are two extreme cases:

```c
        /* 216 - 10240x4320@100Hz 64:27 */
        { DRM_MODE("10240x4320", DRM_MODE_TYPE_DRIVER, 5940000, 10240, 12432,
                   12608, 13200, 0, 4320, 4336, 4356, 4500, 0,
                   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC),
          .picture_aspect_ratio = HDMI_PICTURE_ASPECT_64_27, },
        /* 217 - 10240x4320@120Hz 64:27 */
        { DRM_MODE("10240x4320", DRM_MODE_TYPE_DRIVER, 5940000, 10240, 10528,
                   10704, 11000, 0, 4320, 4336, 4356, 4500, 0,
                   DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC),
          .picture_aspect_ratio = HDMI_PICTURE_ASPECT_64_27, }
```

Reply:
Isn't `1000000LL` already `int64_t`? Does this change actually change anything?

The same with the changes above, isn't `1000.0` already `double`? Don't C promotion rules make this patch a no-op?